batch/batch/front_end/validate.py

```diff
@@ -8,6 +8,7 @@
 #   'always_run': bool,
 #   'attributes': {str: str},
 #   'command': [str],
+#   'shell': str,
```

Comment:
`shell` should be a property of the high-level batch client (hailtop.batch), not the REST API.

Reply:
One more thing, so you're aware: all information that's stored to disk must be versioned so the schema can evolve while future readers still know how to interpret it.  The config goes into the database, and therefore must be versioned and this change would require a version bump.

There is also a multi-step protocol for rolling out new versions: if you add a new field, you first have to commit support for it, deploy, verify all the workers have upgraded, and then you can enable the new feature.  Otherwise, an old worker could receive a config with `shell` that it doesn't handle.

Reply:
Thanks, I didn't realize we versioned job spec, though in retrospect this could have been "obvious" as we have versioned rest endpoints.

Reply:
Well, the endpoints are all currently alpha so subject to change.  We'll change that when they are stable enough and there's enough user pressure.  We're probably close.